lib/ex_machina/ecto.ex

```diff
@@ -79,34 +79,60 @@ defmodule ExMachina.Ecto do
   end
 
   defp get_assocs(%{__struct__: struct}) do
-    for a <- struct.__schema__(:associations) do
-      {a, struct.__schema__(:association, a)}
+    for assoc <- struct.__schema__(:associations) do
+      {assoc, struct.__schema__(:association, assoc)}
     end
   end
 
-  defp belongs_to_assocs(model) do
-    for {a, %{__struct__: Ecto.Association.BelongsTo}} <- get_assocs(model), do: a
+  defp assoc_keys(record) do
+    for {key, _assoc} <- get_assocs(record), do: key
   end
 
-  defp not_loaded_assocs(model) do
-    for {a, %{__struct__: Ecto.Association.Has}} <- get_assocs(model),
-      !Ecto.assoc_loaded?(Map.get(model, a)),
-      do: a
+  defp put_assoc_changes(changeset, record) do
+    keys = assoc_keys(record) -- belongs_to_assoc_keys(record)
+
+    Enum.reduce(keys, changeset, fn(key, changes) ->
+      case Map.get(record, key) do
+        %Ecto.Association.NotLoaded{} ->
```

Comment:
Can you use the struct form like on line 88? That will make CI pass. Ecto is not an explicit dependency so we can't rely on it always being there